master/pkg/schemas/expconf/environment_config.go

```diff
@@ -25,13 +25,14 @@ func (p PodSpec) Copy() interface{} {
 func (p PodSpec) Merge(other interface{}) interface{} {
 	out := k8sV1.Pod{}
 	k8sP := k8sV1.Pod(p)
-	k8sOther := k8sV1.Pod(other.(PodSpec))
+	k8sOther := k8sV1.Pod(*other.(*PodSpec))
 	// Copy the low-priority values first.
 	k8sOther.DeepCopyInto(&out)
 	// Overwrite the object with high-priority values.
 	// DeepCopyInto will only copy non-nil values, so this will effectively merge the objects.
 	k8sP.DeepCopyInto(&out)
-	return PodSpec(out)
+	output := PodSpec(out)
+	return &output
```

Comment:
Something is wrong with this.  A `func (p PodSpec) Merge` _must_ return a plain `PodSpec`, not a `*PodSpec`.  Otherwise, keeping track of what functions returned what types would be a terrible mess.

What arose that caused you to need this?

Reply:
Was testing on a stale copy of my code. We're good